### DISCUSSION TITLE
Cookies persisted across requests

### DISCUSSION THREAD

**Discussion by morcoteg** [gavv/httpexpect#128] [Q&A] [answered]
I'm noticing that in my tests, any cookies that were set previously, are leaking into other requests. The problem is that I have some requests that need the cookies to be empty.

I've worked around this by initializing a new httpexpect object in tests where I don't want any prior requests' cookies set.

The workaround:
```
// Some request is made that sets cookies

...

// Initialize new httpexpect object so cookies from previous request aren't used
e := httpexpect.new(t, server.URL)
// make request using e.POST, cookies won't be set from any prior requests
```

Is there a recommended best practice for this sort of case?

**Comment by gavv** [comment, marked as answer]
Hi, did you try to create a httpexpect object with `Client.Jar` set to `nil`, as described here?

https://github.com/gavv/httpexpect#session-support

This should disable cookies completely.

Or do you need to disable cookies on per-request basis?

**Comment by gavv** [comment]
You can also create different http.Client instances, e.g. with and without cookie jar, and select a different client for each request using WithClient method.

**Comment by morcoteg** [comment]
In this case, I need cookies disabled on a per request basis. Thank you, I'll take a look at that/try that

**Comment by gavv** [comment]
Closing, feel free to reopen if needed.
